.github/workflows/continuous-integration-osx.yml

```diff
@@ -39,9 +48,8 @@ jobs:
             -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF
             -DKokkos_ENABLE_TESTS=On
             -DCMAKE_BUILD_TYPE=${{ matrix.cmake_build_type }}
+            -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
       - name: build
-        run:
-          cmake --build build --parallel 2
+        run: cmake --build build --parallel $(sysctl -n hw.ncpu)
```

Comment:
How much confidence do we have that `sysctl -n hw.ncpu` did what we want?

Reply:
Should we just use `3` if we go by what https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories says for the number of processors on macOS (M1)

Reply:
We have used that in VOTCA for decades https://github.com/votca/votca/blob/22618c7a6dedd59bb2b3296d39afca1b78e50087/csg/share/scripts/inverse/functions_common.sh#L580), but I am fine with either.

I just prefer to not hard-coded things as the runner always change:
https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

Reply:
Fine by me if you have experience with it.